src/librustc_driver/driver.rs

```diff
@@ -1167,15 +1158,11 @@ pub fn phase_5_run_llvm_passes(sess: &Session,
 
 /// Run the linker on any artifacts that resulted from the LLVM run.
 /// This should produce either a finished executable or library.
-#[cfg(feature="llvm")]
-pub fn phase_6_link_output(sess: &Session,
-                           trans: &trans::CrateTranslation,
+pub fn phase_6_link_output<T: TransCrate>(sess: &Session,
```

Comment:
nit: style is to give "significant" type parameters a significant name, e.g. call them `TRANS` instead of `T`.